pkg/runtime/types.go

```diff
@@ -0,0 +1,94 @@
+// Copyright Nitric Pty Ltd.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package runtime
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"path/filepath"
+	"strings"
+
+	"github.com/docker/docker/api/types/mount"
+
+	"github.com/nitrictech/boxygen/pkg/backend/dockerfile"
+)
+
+type Runtime interface {
+	DevImageName() string
+	ContainerName() string
+	FunctionDockerfile(funcCtxDir, version, provider string, w io.Writer) error
+	FunctionDockerfileForCodeAsConfig(w io.Writer) error // FunctionDockerfileForCodeAsConfig generates a base image for code-as-config
+	LaunchOptsForFunction(stackDir string) (LaunchOpts, error)
+	LaunchOptsForFunctionCollect(stackDir string) (LaunchOpts, error)
+}
+
+type RuntimeExt string
+
+const (
+	RuntimeTypescript RuntimeExt = "ts"
+	RuntimeJavascript RuntimeExt = "js"
+	RuntimePython     RuntimeExt = "py"
+	RuntimeGolang     RuntimeExt = "go"
+	RuntimeJava       RuntimeExt = "java"
```

Comment:
As mentioned before for java this will be pointed at a `jar` (too much variability in build and dependency systems to reliably build/run from sources.

Reply:
OK, I suspect java support will need to be fixed up. This change is just moving the existing stuff around, not functionally changing it.

Reply:
no worries, lets look at this in a separate PR.